### ISSUE TITLE
`remark-lint-no-undefined-references` is flagging GFM checkboxes

### ISSUE THREAD

**Issue by JulianCataldo** [remarkjs/remark-lint#295]
### Initial checklist

- [X] I read the [support docs](https://github.com/remarkjs/.github/blob/main/support.md)
- [X] I read the [contributing guide](https://github.com/remarkjs/.github/blob/main/contributing.md)
- [X] I agree to follow the [code of conduct](https://github.com/remarkjs/.github/blob/main/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Aremarkjs&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

remark-preset-lint-recommended@latest

### Steps to reproduce

```js
  const output = await remark()
    .use(remarkPresetLintRecommended)
    .use(remarkPresetLintMarkdownStyleGuide)
    .use(remarkPresetLintConsistent)
    .process(content);
```

### Expected behavior

```
- [x] Salt
```

### Actual behavior

![Screenshot 2022-09-02 at 0 49 45](https://user-images.githubusercontent.com/603498/188025625-d01b3167-34b5-43ab-9e14-bed4eaf0d4a4.png)

**Comment by JounQin** [comment]
You didn't include `remark-gfm`.

**Comment by JulianCataldo** [comment]
Yes thank you 👍

I've just noticed this, as I was using two separate pipelines, one for **transformation**, and one for **linting** only.
